cypress/e2e/models/migration/dynamic-report/dependencies/dependencies.ts

```diff
@@ -73,12 +73,15 @@ export class Dependencies {
             Dependencies.validateFilter(dependency);
         });
 
-        if (dependenciesNotExpected.length > 0) {
-            getUniqueElementsFromSecondArray(dependenciesExpected, dependenciesNotExpected).forEach(
-                (dependency: AppDependency) => {
+        if (dependenciesNotExpected) {
+            if (dependenciesNotExpected.length > 0) {
```

Comment:
This can be merged into a single  if:

``` typescript 
  if (dependenciesNotExpected && dependenciesNotExpected.length) 
```

Reply:
Tried, it failed cause if `dependenciesNotExpected` doesn't exist and it tried to calculate its length - it failed with an exception

Reply:
As we discussed, the functionality would be exactly the same one, as this is a matter of preference and we don't have a style guide your solution is correct as it is. Approving the PR then.

Reply:
I agree with @abrugaro comment . see if this can be improved .

Reply:
Applied